cdap-data-fabric/src/main/java/io/cdap/cdap/store/StoreDefinition.java

```diff
@@ -179,6 +181,74 @@ public static void create(StructuredTableAdmin tableAdmin) throws IOException {
     }
   }
 
+  /**
+   * Schema for SourceControlMetadata table.
+   */
+  public static final class SourceControlMetadataStore {
+
+    public static final StructuredTableId SOURCE_CONTROL_METADATA = new StructuredTableId(
+        "source_control_metadata");
+
+    public static final String NAMESPACE_FIELD = "namespace";
+    public static final String TYPE_FIELD = "type";
+    public static final String NAME_FIELD = "name";
+    public static final String SPECIFICATION_HASH_FIELD = "specification_hash";
+    public static final String COMMIT_ID_FIELD = "commit_id";
+    public static final String LAST_MODIFIED_FIELD = "last_modified";
+    public static final String IS_SYNCED_FIELD = "is_synced";
+
+    public static final StructuredTableSpecification SOURCE_CONTROL_METADATA_TABLE_SPEC =
+        new StructuredTableSpecification.Builder()
+        .withId(SOURCE_CONTROL_METADATA)
+        .withFields(Fields.stringType(NAMESPACE_FIELD),
+            Fields.stringType(TYPE_FIELD),
+            Fields.stringType(NAME_FIELD),
+            Fields.stringType(SPECIFICATION_HASH_FIELD),
+            Fields.stringType(COMMIT_ID_FIELD),
+            Fields.longType(LAST_MODIFIED_FIELD),
+            Fields.booleanType(IS_SYNCED_FIELD))
+        .withPrimaryKeys(NAMESPACE_FIELD, TYPE_FIELD, NAME_FIELD)
+        .withIndexes(IS_SYNCED_FIELD, LAST_MODIFIED_FIELD)
+        .build();
+
+    public static void create(StructuredTableAdmin tableAdmin) throws IOException {
+      createIfNotExists(tableAdmin, SOURCE_CONTROL_METADATA_TABLE_SPEC);
+    }
+  }
+
+  /**
+   * Schema for Repository Metadata table.
+   */
+  public static final class RepositoryMetadataStore {
```

Comment:
What is the difference between RepositoryMetadataStore and SourceControlMetadataStore?

Reply:
SourceControlMetadataStore contains source control metadata (hash and commit Id) and sync status of namespace pipelines.
RepositoryMetadata contains information of all the pipelines in the connected repository. This table was made so that we can have pagination.